drivers/sensor/magntek/magntek_mt6701/magntek_mt6701.c

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright (c) 2024, Vinicius Miguel
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT magntek_mt6701
+
+#include <errno.h>
+#include <zephyr/sys/__assert.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/drivers/spi.h>
+#include <zephyr/logging/log.h>
+#include <math.h>
+#include <zephyr/drivers/sensor/magntek_mt6701.h>
+
+LOG_MODULE_REGISTER(magntek_mt6701, CONFIG_SENSOR_LOG_LEVEL);
+
+#define MT6701_SPI_CONFIG     SPI_OP_MODE_MASTER | SPI_MODE_CPHA | SPI_WORD_SET(8)
+#define MT6701_FULL_ANGLE     360
+#define MT6701_PULSES_PER_REV 4096
+#define MT6701_MILLION_UNIT   1000000
+
+struct mt6701_dev_cfg {
+	struct spi_dt_spec spi_port;
+};
+
+struct mt6701_dev_data {
+	uint32_t position;
+	uint8_t field_status;
+	int8_t push_status;
+	uint8_t loss_status;
+};
+
+static uint8_t table_crc[64] = {0x00, 0x03, 0x06, 0x05, 0x0C, 0x0F, 0x0A, 0x09, 0x18, 0x1B, 0x1E,
+				0x1D, 0x14, 0x17, 0x12, 0x11, 0x30, 0x33, 0x36, 0x35, 0x3C, 0x3F,
+				0x3A, 0x39, 0x28, 0x2B, 0x2E, 0x2D, 0x24, 0x27, 0x22, 0x21, 0x23,
+				0x20, 0x25, 0x26, 0x2F, 0x2C, 0x29, 0x2A, 0x3B, 0x38, 0x3D, 0x3E,
+				0x37, 0x34, 0x31, 0x32, 0x13, 0x10, 0x15, 0x16, 0x1F, 0x1C, 0x19,
+				0x1A, 0x0B, 0x08, 0x0D, 0x0E, 0x07, 0x04, 0x01, 0x02};
+
+static uint8_t mt6701_crc_check(uint32_t input)
+{
+	uint8_t index = 0;
+	uint8_t crc = 0;
+
+	index = (uint8_t)(((uint32_t)input >> 12u) & 0x0000003Fu);
+
+	crc = (uint8_t)(((uint32_t)input >> 6u) & 0x0000003Fu);
+	index = crc ^ table_crc[index];
+
+	crc = (uint8_t)((uint32_t)input & 0x0000003Fu);
+	index = crc ^ table_crc[index];
+
+	crc = table_crc[index];
+
+	return crc;
+}
+
+static int mt6701_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct mt6701_dev_data *dev_data = dev->data;
+	const struct mt6701_dev_cfg *dev_cfg = dev->config;
+
+	uint32_t spi_32 = 0;
+	uint32_t angle_spi = 0;
+	uint8_t received_crc = 0;
+
+	uint8_t read_data[3] = {0, 0, 0};
+
+	const struct spi_buf rx_buf[] = {{
+		.buf = read_data,
+		.len = ARRAY_SIZE(read_data),
+	}};
+
+	const struct spi_buf_set rx = {
+		.buffers = rx_buf,
+		.count = ARRAY_SIZE(read_data),
+	};
+
+	int err = spi_read_dt(&dev_cfg->spi_port, &rx);
+
+	/* an invalid reading preserves the last good value */
+	if (!err) {
```

Comment:
Zephyr by convention branch for handling error, please invert the condition and branch to perform the error handling.